compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -1243,6 +1252,21 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
         }
     }
 
+    fn dectect_wrong_primitive(path: &[Segment]) -> Option<Symbol> {
```

Comment:
I'm not sure the current name is very descriptive. Perhaps something like the following:
```suggestion
    // Returns the name of the Rust type approximately corresponding to
    // a type name in another programming language.
    fn likely_rust_type(path: &[Segment]) -> Option<Symbol> {
```